hpx/util/itt_notify.hpp

```diff
@@ -221,11 +221,23 @@ namespace hpx { namespace util { namespace itt
     //////////////////////////////////////////////////////////////////////////
     struct domain
     {
-        HPX_EXPORT domain(char const* name);
+        HPX_NON_COPYABLE(domain);
+
+        HPX_EXPORT domain(char const*);
+        HPX_EXPORT domain();
+        HPX_EXPORT virtual ~domain();
```

Comment:
Do we really have to make the destructor virtual?

Reply:
Looks like we don't need explicit destructors at all as both types, `domain`, and `thread_domain` currently expose empty destructors only and `thread_domain` doesn't even has members.

Reply:
Good point, `virtual` is not necessary in this case, but default destructors is even better.